src/types/app_config.go

```diff
@@ -46,6 +46,9 @@ func (a AppConfig) GetDependencyNames() []string {
 // GetServiceData returns the configurations data for the given services
 func (a AppConfig) GetServiceData() map[string]ServiceData {
 	result := make(map[string]ServiceData)
+	for serviceName, data := range a.Services.Worker {
```

Comment:
Gahh, I hate that we have three for loops so much in this file. Thoughts on creating a helper function that does the following
```go
func (a AppConfig) forEachService(fn func(string, string, ServiceData)) {
  for serviceName, data := range a.Services.Worker {
    fn("worker", serviceName, data) 
  }
  // similar for private and public
}
```
That would make this method be
```go
func (a AppConfig) GetServiceData() map[string]ServiceData {
  result := make(map[string]ServiceData)
  a.forEachService(func(type, name string, data ServiceData) {
    result[serviceName] = data 
  })
  return result
}
```
Likewise it would simplify quite a few methods in here